packages/user-interface/src/components/Document.tsx

```diff
@@ -16,7 +16,7 @@ const Document = ({ document: doc }: Props) => {
   const { keycloakToken } = useContext(KeycloakContext);
   const { restUri } = useContext(ApiContext);
   const { hrefLang } = useContext(LocaleContext);
-  const downloadLink = `${restUri}/documentapi/${doc.documentapi}/document/${doc.uuid}/content`;
+  const downloadLink = `${restUri}/zakenapi/zaakdocument/${doc.identificatie}/content`;
```

Comment:
Nu werkt dit component alleen bij zaken?

Reply:
Nope, both the [documenten](https://github.com/nl-portal/nl-portal-frontend-libraries/blob/master/packages/api/src/queries/get-documenten.ts) query and zaken query works for this, the main changes were in the backend.

Reply:
I had another look and, we don't even have a separate documenten query.

Reply:
In the future the getProduct should also have documents. How does that work?

Reply:
What is a Product?
If you ask me, an informatieobejct has to have an "owner". In this case the owner is a Zaak and the link betwee na Zaak and Informatieobject is a ZaakInformatieobject. By doing this change we can do validation on whether a user is allowed access to that informatieobject based on what Zaak the informatieobject is linked to.
In the case of a Product, I would assume that the Product will keep track of said link and in which case we need to refactor the current approach regardless.
@nielserkamp-ritense Thoughts?

Reply:
@Borai I am afraid that if a product has documents the verification whether or not a document on a product is viewable by the user is related to the product itself. So it would most likely require a separate endpoint / verifications for products (i.e. /product/{id}/documents/{id}).

The problem we are currently facing is the problem of what is related to what and who has rights to view it. That is currently not forced into the api spec of the portal. Which allows random users to "guess" an document of an other user and view it (something we absolutely do not want in my opinion).

Reply:
My proposal is to make the downloadLink a Prop of Document so that we can pass it into the Document component regardless of where it's used. What do you think?

Reply:
Is this why there was ${doc.documentapi} to let the document object decide where the download is coming from?

Reply:
Partially. it was there because there can be multiple documenten apis connected at the same time (don't know of a usecase for this though), so a download needed to know what instance of documenten api to ask the file content from.
In the new situation we want the "owner" to figure that out behind the scene. all the user has to do is specify which, in this case ZaakDocument but in case of the future Product probably a ProductDocument, they want to download.

Reply:
So it's better to not specify this path inside the FE code. Who do we know what api to call?

Reply:
That sounds like a bigger design question and a bit too much for this PR.

The problem was: anyone who is authorized can download any document regardless of it's status, vertrouwlijkheidsaanduiding or whether it's even linked to a case or not as long as they can guess the documentenapi config name and a uuid.

https://github.com/nl-portal/nl-portal-backend-libraries/pull/267 is now approved and waiting on this PR. We would like these merged because Amsterdam Wonenportaal wants this safety feature in for Digid certification.

Reply:
Understood :) 

Can we for now change the props to this and send the whole link to this component. 

```suggestion
interface Props {
  downloadLink: String;
}
```

Reply:
Yess, will do